### DISCUSSION TITLE
Is it possible to support dialect as option

### DISCUSSION THREAD

**Discussion by shengha2** [agronholm/sqlacodegen#202] [Q&A] [answered]
We are generating our python code using sqlacodegen so far it works really well.  But recently, we bump into an issue with only the Postgres dialect ARRAY support feature that could not be supported with sqlalchemy ARRAY.  

We have (generated from generator.py)
`from sqlalchemy import ARRAY`

We need
`from sqlalchemy.dialects.postgresql import ARRAY`

Is it possible to specify a flag for generating the postgresql dialect?

**Comment by leonarduschen** [comment, marked as answer]
Quoting from `sqlalchemy.ARRAY` official [documentation](https://docs.sqlalchemy.org/en/14/core/type_basics.html#sqlalchemy.types.ARRAY)

>This type serves as the basis for all ARRAY operations. However, currently only the PostgreSQL backend has support for SQL arrays in SQLAlchemy. It is recommended to use the PostgreSQL-specific [sqlalchemy.dialects.postgresql.ARRAY](https://docs.sqlalchemy.org/en/14/dialects/postgresql.html#sqlalchemy.dialects.postgresql.ARRAY) type directly when using ARRAY types with PostgreSQL, as it provides additional operators specific to that backend.

@agronholm seems like we can just replace `from sqlalchemy import ARRAY` with `from sqlalchemy.dialects.postgresql import ARRAY`

**Reply by leonarduschen** [reply]
On a second thought...

Current logic for column type is pretty straightforward: choose the most generic type that still has the same compiled SQL expression (with the special case of Postgresql's float), users who wish to use dialect specific types will have to make manual changes.

Manually changing `from sqlalchemy import ARRAY` with `from sqlalchemy.dialects.postgresql import ARRAY` in the generated code is easy and IMO doesn't warrant adding an extra special case despite what the docs says above.

Adding the option to use dialect specific types instead of generic types sounds like a good idea to me though, implementation will be quite straightforward as this is already the default behavior of `sqlalchemy.MetaData.reflect`

**Reply by shengha2** [reply]
Thank you for the thoughtful solutions proposed here. One thing that is hard in our case is that we maintain a large project and regenerate the orm via github action. Sometimes we can do many code generations in a day. Manually changing from `sqlalchemy import ARRAY` with from `sqlalchemy.dialects.postgresql import ARRAY` would not be feasible in our case.

**Reply by leonarduschen** [reply]
Thanks for elaborating on your usecase. I'll create an issue for  an option to generate dialect-specific column types.

**Reply by shengha2** [reply]
Thank you
